src/Symfony/Component/DependencyInjection/Compiler/InlineServiceDefinitionsPass.php

```diff
@@ -148,7 +148,7 @@ private function isInlineableDefinition(ContainerBuilder $container, $id, Defini
             return false;
         }
 
-        if (count($ids) > 1 && $definition->getFactoryService(false)) {
+        if (count($ids) > 1 && $definition->getFactory()) {
```

Comment:
this should be removed entirely instead. The case of factories is already covered by the previous condition